### PR TITLE
add darwin/osx support to slave class

### PR DESCRIPTION
**Pull request by adamcstephens**
This utilizes the existing defaults file, along with a wrapper script and launchd service. There is no wget by default on OSX so use curl there (could be used everywhere, or instead use a puppet module such as archive).

### CODE REVIEW

File: spec/classes/jenkins_slave_spec.rb [hunk 1]
```diff
@@ -67,6 +68,19 @@
     end
   end
 
+#  describe 'Darwin' do
```
Comment (rtyler) [inline spec/classes/jenkins_slave_spec.rb]: Is there any reason for these examples to be commented out?

File: manifests/slave.pp [hunk 2]
```diff
@@ -145,31 +114,110 @@
...
 
-  file { '/etc/init.d/jenkins-slave':
-    ensure => 'file',
-    mode   => '0755',
-    owner  => 'root',
-    group  => 'root',
-    source => "puppet:///modules/${module_name}/jenkins-slave.${::osfamily}",
-    notify => Service['jenkins-slave'],
+  case $::kernel {
+    'Linux': {
```
Comment (rtyler) [inline manifests/slave.pp]: This case statement smells a bit to me, would it be possible to refactor this into a `jenkins::slave::linux` and a `jenkins::slave::darwin` module which would then be included depending on the kernel?

**Comment by rtyler** [comment]
@adamcstephens overall, looking very good! some minor comments that I've added; can't wait to get this merged

File: spec/classes/jenkins_slave_spec.rb (hunk 1 above)
Comment (adamcstephens) [inline spec/classes/jenkins_slave_spec.rb]: I'm still learning rspec, and can't figure out how to get the tests to ignore java install on darwin.

```
Failures:

  1) jenkins::slave Darwin with slave_name behaves like using slave_name should contain File[/home/jenkins-slave/jenkins-slave] with content =~ /^CLIENT_NAME="jenkins-slave"$/
     Failure/Error: it { should contain_file(slave_runtime_file).with_content(/^CLIENT_NAME="jenkins-slave"$/) }
     Puppet::Error:
       Could not find resource 'Class[Java]' for relationship on 'Service[jenkins-slave]' on node myosx.local
     Shared Example Group: "using slave_name" called from ./spec/classes/jenkins_slave_spec.rb:80
     # ./spec/classes/jenkins_slave_spec.rb:43:in `block (3 levels) in <top (required)>'
```

**Comment by rtyler** [comment]
I'm going to merge this and try to clean it up a bit myself before release

File: spec/classes/jenkins_slave_spec.rb (hunk 1 above)
Comment (rtyler) [inline spec/classes/jenkins_slave_spec.rb]: so the puppetlabs/java module doesn't support Darwin so I think part of why this is failing.

**Comment by adamcstephens** [comment]
Ok, sorry. I started on the refactor earlier in the week but work got the better of me.

**Comment by rtyler** [comment]
@adamcstephens no big deal, this has since been released :)

**Pull request merged**